pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -327,6 +330,23 @@ func verifyRequiredMinishiftVersion(meta addon.AddOnMeta) error {
 	return nil
 }
 
+func (m *AddOnManager) verifyRequiredAddons(metadata addon.AddOnMeta) error {
+	var depsNotInstalled []string
+	dependencies, err := metadata.Dependency()
+	if err == nil && len(dependencies) > 0 {
+		for _, dep := range dependencies {
+			if m.IsInstalled(dep) {
+				continue
+			}
+			depsNotInstalled = append(depsNotInstalled, dep)
+		}
+		if len(depsNotInstalled) != 0 {
+			return errors.New(fmt.Sprintf("The add-on depends on: [%s]. Please install these first.", strings.Join(depsNotInstalled, ", ")))
```

Comment:
`Please install these first` is technically wrong as installing some dependent add-on is not enough for making sure these add-ons were applied before applying an add-on. So wondering how can we communicate this properly.

Reply:
suggestion `fmt.Sprintf("Dependent add-ons [%s] not found for this instance. Please install and apply them before running this add-on.", strings.Join(depsNotInstalled, ", "))`